### PR TITLE
Added two differents errors events. 'error' is for specific AWS clien…

### PR DESCRIPTION
**Pull request by pablovilas**
…t error and 'processing_error' is for app message processing errors

All test are passing, documentation updated. What do you think?
Regards

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,15 +10,24 @@ var requiredOptions = [
...
+  this.name = 'SQSError';
+  this.message = (message || '');
+}
+SQSError.prototype = Error.prototype;
+
 function validate(options) {
   requiredOptions.forEach(function (option) {
     if (!options[option]) {
-      throw new Error('Missing SQS consumer option [' + option + '].');
+      throw new SQSError('Missing SQS consumer option [' + option + '].');
```
Comment (robinjmurphy) [inline index.js]: I think these can remain as just plain `Error`'s as they're thrown and not handled by any events.

**Comment by robinjmurphy** [comment]
Looking good. Can we get a test for when the delete message/receive message fails. Make sure that it fires an `error` event.

File: index.js (hunk 1 above)
Comment (pablovilas) [inline index.js]: I'll change that to Error

**Comment by robinjmurphy** [comment]
My bad. These tests already exist. Just the change to the validation errors and then I'll get this merged and released tomorrow.

**Comment by pablovilas** [comment]
Sure. Right away

**Comment by pablovilas** [comment]
Closing this pr and opening a new one.

**Pull request closed without merging**